File: backend/src/main/scala/shared/http/response/SuccessResponse.scala
package shared.http.response

import shared.model.ApiMessageParams
import shared.http.codec.SharedHttpCodecs.given
import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class SuccessResponse(
  code: Option[String],
  message: Option[String],
  params: ApiMessageParams
)

object SuccessResponse:
  given Encoder[SuccessResponse] = deriveEncoder[SuccessResponse]
  given Decoder[SuccessResponse] = deriveDecoder[SuccessResponse]

File: frontend/src/objects/shared/response/SuccessResponse.ts
import type { ApiMessageParams } from '@/objects/shared/ApiMessageParams'

/** API 成功响应；用于无业务实体返回的命令结果，可附带提示消息参数。 */
export type SuccessResponse = {
  code: string | null
  message: string | null
  params: ApiMessageParams
}
